lib/grant-types/abstract-grant-type.js

```diff
@@ -0,0 +1,95 @@
+
```

Comment:
Not sure I understand. We are validating the client when [generating a token](https://github.com/thomseddon/node-oauth2-server/pull/203/files#diff-f3ac454512815b3440a11af47dc3412cR110) and [generating an authorization code](https://github.com/thomseddon/node-oauth2-server/pull/203/files#diff-5b2d9d61bd65477fa282ec9faecc4becR147).

There is no validation of the client when authenticating using an access token because [the spec](https://tools.ietf.org/html/rfc6750#section-2.1) does not require it. An access token is sufficient by itself.